### PR TITLE
Limit the number of operations in a single json patch to be 10,000

### PR DESCRIPTION
**Pull request by soltysh**
This PR is actually a set of fixes brought into 1.12 in https://github.com/kubernetes/kubernetes/pull/74104. Additionally it fixes `glide.yaml` which was not updated in https://github.com/openshift/origin/pull/22045 properly with kube-openapi sha. Finally since I touched deps I bumped all the deps to replace https://github.com/openshift/origin/pull/22238 which had issues with kube-openapi. 

/assign @deads2k @sttts @sjenning @enj

### CODE REVIEW

**Comment by deads2k** [comment]
/hold

still messing with component-base.

**Comment by soltysh** [comment]
/hold cancel
the component-base is now fixed

File: glide.lock [hunk 1]
```diff
@@ -1,5 +1,5 @@
-hash: f673b06826e9fa526bda285c5109b133989e188db2d8aca9dbc6913af034857d
-updated: 2019-02-19T14:54:01.742954187-05:00
+hash: 38c293cd395b48a47f66c9434e2e7b1b19aa388c37259a06d772e6ceaca4c1cc
```
Comment (p0lyn0mial) [inline glide.lock]: not sure if this is correct as this `hash` might correspond to the content of `glide.yaml` which might be different for the `master` branch.

**Pull request merged**